### PR TITLE
[variant] Add Generic F7xx

### PR DESCRIPTION
**Pull request by ABOSTM**
**Summary**
[variant] Add Generic F7xx

Fixes #1302

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -232,6 +233,24 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
...
+| :yellow_heart: | STM32F746ZE<br>STM32F746ZG | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F756ZG | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F750Z8 | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F765ZG<br>STM32F765ZI | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F767ZG<br>STM32F767ZI | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F777ZI | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F765VG<br>STM32F765VI | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F767VG<br>STM32F767VI | Generic Board | **2.0.0** |  |
+| :yellow_heart: | STM32F777VI | Generic Board | **2.0.0** |  |
+
```
Comment (fpistm) [inline README.md]: Should be in alphabetical order

File: variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/generic_clock.c [hunk 2]
```diff
@@ -24,8 +24,52 @@
   */
 WEAK void SystemClock_Config(void)
 {
-  /* SystemClock_Config can be generated by STM32CubeMX */
-#warning "SystemClock_Config() is empty. Default clock at reset is used."
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
```
Comment (fpistm) [inline variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/generic_clock.c]: remove the 0.
```suggestion
  RCC_OscInitTypeDef RCC_OscInitStruct = {};
  RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
```

File: variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/ldscript.ld [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  _sidata = LOADADDR(.data);
+
+  /* Initialized data sections into "RAM" Ram type memory */
+  .data :
+  {
+    . = ALIGN(4);
+    _sdata = .;        /* create a global symbol at data start */
+    *(.data)           /* .data sections */
+    *(.data*)          /* .data* sections */
+
```
Comment (fpistm) [inline variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/ldscript.ld:135]: *(.RamFunc)        /* .RamFunc sections */
    *(.RamFunc*)       /* .RamFunc* sections */

File: variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/generic_clock.c [hunk 4]
```diff
@@ -23,8 +23,52 @@
   */
 WEAK void SystemClock_Config(void)
 {
-  /* SystemClock_Config can be generated by STM32CubeMX */
-#warning "SystemClock_Config() is empty. Default clock at reset is used."
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
```
Comment (fpistm) [inline variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/generic_clock.c]:
```suggestion
  RCC_OscInitTypeDef RCC_OscInitStruct = {};
  RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
```

File: variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/ldscript.ld [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  _sidata = LOADADDR(.data);
+
+  /* Initialized data sections into "RAM" Ram type memory */
+  .data :
+  {
+    . = ALIGN(4);
+    _sdata = .;        /* create a global symbol at data start */
+    *(.data)           /* .data sections */
+    *(.data*)          /* .data* sections */
+
```
Comment (fpistm) [inline variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/ldscript.ld:135]: *(.RamFunc)        /* .RamFunc sections */
    *(.RamFunc*)       /* .RamFunc* sections */

File: variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/generic_clock.c [hunk 6]
```diff
@@ -24,8 +24,54 @@
   */
 WEAK void SystemClock_Config(void)
 {
-  /* SystemClock_Config can be generated by STM32CubeMX */
-#warning "SystemClock_Config() is empty. Default clock at reset is used."
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
```
Comment (fpistm) [inline variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/generic_clock.c]:
```suggestion
  RCC_OscInitTypeDef RCC_OscInitStruct = {};
  RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
```

File: variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/ldscript.ld [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+  _sidata = LOADADDR(.data);
+
+  /* Initialized data sections into "RAM" Ram type memory */
+  .data :
+  {
+    . = ALIGN(4);
+    _sdata = .;        /* create a global symbol at data start */
+    *(.data)           /* .data sections */
+    *(.data*)          /* .data* sections */
+
```
Comment (fpistm) [inline variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/ldscript.ld:135]: *(.RamFunc)        /* .RamFunc sections */
    *(.RamFunc*)       /* .RamFunc* sections */

File: variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/generic_clock.c [hunk 8]
```diff
@@ -22,8 +22,54 @@
   */
 WEAK void SystemClock_Config(void)
 {
-  /* SystemClock_Config can be generated by STM32CubeMX */
-#warning "SystemClock_Config() is empty. Default clock at reset is used."
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
```
Comment (fpistm) [inline variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/generic_clock.c]:
```suggestion
  RCC_OscInitTypeDef RCC_OscInitStruct = {};
  RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
  RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {};
```

File: variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/ldscript.ld [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+  _sidata = LOADADDR(.data);
+
+  /* Initialized data sections into "RAM" Ram type memory */
+  .data :
+  {
+    . = ALIGN(4);
+    _sdata = .;        /* create a global symbol at data start */
+    *(.data)           /* .data sections */
+    *(.data*)          /* .data* sections */
+
```
Comment (fpistm) [inline variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/ldscript.ld:135]: *(.RamFunc)        /* .RamFunc sections */
    *(.RamFunc*)       /* .RamFunc* sections */

**Review by fpistm — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ABOSTM) [reply README.md]: Done

File: variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/generic_clock.c (hunk 2 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/generic_clock.c]: Done

File: variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/ldscript.ld (hunk 3 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F745Z(E-G)Tx_F746Z(E-G)Yx_F746ZETx_F746ZGTx_F750Z8Tx_F756ZG/ldscript.ld:135]: Done

File: variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/generic_clock.c (hunk 4 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/generic_clock.c]: Done

File: variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/ldscript.ld (hunk 5 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F746B(E-G)Tx_F746NEHx_F746NGHx_F750N8Hx_F756BGTx_F756NGHx/ldscript.ld:135]: Done

File: variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/generic_clock.c (hunk 6 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/generic_clock.c]: Done

File: variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/ldscript.ld (hunk 7 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F765V(G-I)_F767VG_F767VI_F777VI/ldscript.ld:135]: Done

File: variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/generic_clock.c (hunk 8 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/generic_clock.c]: Done

File: variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/ldscript.ld (hunk 9 above)
Comment (ABOSTM) [reply variants/STM32F7xx/F765Z(G-I)Tx_F767ZGTx_F767ZITx_F777ZITx/ldscript.ld:135]: Done

**Review by fpistm — APPROVED** [review]
LGTM. Will integrate to offciial PR
Thanks.

**Comment by fpistm** [comment]
Integrated to official PR.

Thanks

**Pull request closed without merging**